DATA: Add Shelter Model

Spectralgo opened the pull request:
Closes #32

ElbekDeveloper commented:
Please,  add migrations

rcruzfreelance commented:
@Spectralgo  @ElbekDeveloper  I could add the migrations If you let me

Spectralgo commented:
@rcruzfreelance go ahead :)

rcruzfreelance commented:
@Spectralgo @ElbekDeveloper theres seem to be some issue with the migration since our model has IEnumerable which is not acceptable for sql server that I know of.

![image](https://user-images.githubusercontent.com/53588704/169428635-9d3c22fa-25b1-4b70-8189-baca71380a77.png)

We can either create seperate class for these or use enums to identify they types. 

What are your thoughts?

ElbekDeveloper commented on RefugeeLand.Core.Api/Models/Shelters/Shelters.cs:
```diff
@@ -0,0 +1,35 @@
+    {
+        public Guid Id { get; set; }
+        public Guid HostId { get; set; }
+        public string ShelterType { get; set; }
```
Maybe this should be `enum` @rcruzfreelance ?

ElbekDeveloper commented:
@rcruzfreelance conflicts

Spectralgo commented:
![Screenshot_20220520-102320_GitHub.jpg](https://user-images.githubusercontent.com/56794425/169466152-d2dfcf5d-26e6-46da-9da0-db36e81e619e.jpg)

This branch should be updated/merged with the main branch.

You can see what I did for the languages.

1. I updated/merged the branch and resolve the conflicts

2. I converted Languages from a string into a IList

3. I added migrations

rcruzfreelance commented:
Allright cool i can check it out thanks @Spectralgo

rcruzfreelance commented:
@Spectralgo I checked and made the changes and it still giving out error. IList is also not supported, here is the error I get

![image](https://user-images.githubusercontent.com/53588704/169534020-2bd62c08-ed1a-4654-a12f-16a883709792.png)

Languages should either be a string,enum or a class since Sql server do not support list data types.

Here is a link as a reference to show what I mean

https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-data-type-mappings

rcruzfreelance commented:
@Spectralgo  Im still getting error from migrations here is the error
![image](https://user-images.githubusercontent.com/53588704/169535681-8525c7d0-c7ad-4c73-a474-c807f2ca2d0b.png)

Since were still using Ilist for languages is not going to be supported for Sql Server. It should either a string, enum or a Class in itself

Here is a reference of what I mean
https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-data-type-mappings

Let me know on your thoughts on this

rcruzfreelance commented:
@Spectralgo thanks for the clarifications. Was able to solve the conflict and created the migration

ElbekDeveloper commented on RefugeeLand.Core.Api/Migrations/20220520211059_AddShelterModel.cs:
```diff
@@ -0,0 +1,19 @@
+    public partial class AddShelterModel : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
+        }
```
I don't believe that this is going to create any tables. Can somebody explain how our shelter is being migrated. Maybe we forgot to put `DbSet<Shelter>` in the `StorageBroker.Shelters.cs` because I didn't see the changes... cc: @rcruzfreelance  @Spectralgo

Spectralgo commented:
@ElbekDeveloper is right. We could just add it, rebuild the migration, and call it a day, but the commit history won't be clean enough. 

We should remake a branch from scratch in my opinion.

rcruzfreelance commented:
Ah youre right I can fix this let try it again and you guys let me know after

rcruzfreelance commented:
The migrations can still be intact if we use the right commands for it(dotnet ef migrations remove) I added the DbSet for Shelter(sorry about that) ,I tried to fix it it, and it still gave me an error for Shelter since is using AllowedPets which is an IEnumerable<Pet> which is still not supported for sql server. 

![image](https://user-images.githubusercontent.com/53588704/169660117-6843f417-81dc-4b1a-b3b4-c97bde38d9e6.png)

I would also agree with @Spectralgo and starting over again.

Spectralgo commented:
@rcruzfreelanceI banged my head on migrations all day ... we need to have a plan :p

Spectralgo commented:
@rcruzfreelance https://youtu.be/R_-QW3AJb98

rcruzfreelance commented:
The only problem I see with this particular migrations is that were adding IEnumerable of a enum to the model. If it we use a normal enum Pets(without IEnumerable) it would work but having the IEnumerable is like telling the relationship between tables which we know thats not the case. If we were to add a Pet class instead of an enum then we use it in the Db Set the IEnumerable <with Pet class>, the migration can work

Spectralgo commented:
You are right. We should have a pet class, we should talk about it, I would like to tell you what I was trying to do by choosing an enum. But you are totally right it's not the right thing to do. And it should be a IList instead of a IEnum

rcruzfreelance commented:
sure I have some free time although having to go through a 9 hour session must be exhausting lol. Let me know what other day you are free. For me it is 1:07PM in Puerto Rico so im guessing were a few hours apart

Sharaf-Mansour commented on RefugeeLand.Core.Api/Migrations/StorageBrokerModelSnapshot.cs:
```diff
@@ -1,12 +1,9 @@
-// -------------------------------------------------------
-
-// <auto-generated />
+﻿// <auto-generated />
```
Missing header

ElbekDeveloper replied:
You don't need header file when you see
`// <auto-generated />`

Sharaf-Mansour commented on RefugeeLand.Core.Api/Migrations/20220520211059_AddShelterModel.Designer.cs:
```diff
@@ -0,0 +1,153 @@
+﻿// <auto-generated />
```
Missing header

Sharaf-Mansour commented on RefugeeLand.Core.Api/Migrations/20220520211059_AddShelterModel.cs:
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Missing header

ElbekDeveloper commented:
Is this still relevant @Spectralgo ?

Spectralgo commented:
Nope
